### DISCUSSION TITLE
Redstone signal

### DISCUSSION THREAD

**Discussion by TheWasabiDude** [GriefPrevention/GriefPrevention#647] [Q&A] [answered]
I like how you thought of almost every possible way to protect a house - like water not flowing into it, piston being unable to push a block inside it and so on.

But there is one more thing. Redstone signal shouldn't transfer between unprotected and protected areas, because right now if there is an iron door or trapdoor or anything at the edge of the protected area, you can just place a redstone torch or redstone block outside and it will open.

Not a huge deal as it only gives you access into the area and won't lent you grief anything there if your config is set up correctly, but still not something that should happen.

Another thing is as far as I saw buttons are not protectable (or if so, it's not working for us) and if there is a wood button inside of a protected wall, you can press it by repeatedly throwing a trident to that wall because eventually it will glitch through it and press the button on it's way back (with Loyalty enchant)

**Comment by bigpresh** [comment]
I think this (blocking redstone signals into a claim) has been suggested before, but would be quite expensive performance-wise to implement.  Players can trivially avoid this happening by just making their claim big enough to protect what they want to protect, i.e. extend outwards at least another block beyond their walls.

Not seen the glitch of getting tridents to trigger a button on the reverse of a wall, though, that's an interesting one.  Do you need to standing up against the wall for it to work?

**Comment by TomLewis** [comment]
This is more teaching players to just build their claim big enough.

**Comment by Jikoo** [comment]
Regarding redstone, I agree with everyone else - it's an incredibly costly check that can be avoided by teaching players to claim and build using common sense. There are so many situations that are very hard to check for, a LOT of intense block examination during every very frequent redstone event.

That's a neat trident exploit, don't think it would be able to be patched out by changing spawn location to keep the hitbox in the same place due to how the return angle can be changed.
In general, I think it's good for GP to keep its button/pressure plate usage consistent, but an addon to change that behavior would be good.

**Comment by TheWasabiDude** [comment]
I fail to see how not protecting pressure plates and especially buttons consistent, it always seemed like a "thats weird, oh well" kind of thing. Everything should be protected with a switch in the config file to turn off if you wish

**Comment by Jikoo** [comment]
>I fail to see how not protecting pressure plates and especially buttons consistent, it always seemed like a "thats weird, oh well" kind of thing. Everything should be protected with a switch in the config file to turn off if you wish

Re: buttons in general - levers and stone buttons require access trust, assuming your configuration is set up for that.

I can't find it now, but I believe BigScary commented on pressure plates in the past. It was an intentional design decision due to being usable by any living entity. In addition, item entities are able to trigger non-stone pressure plates. Wooden button usage by projectiles is actually not possible to directly detect via Spigot's API, you have to guess based on which entities are nearby. It becomes a mess - under what circumstances do we block access by other entities?

Like I said, it would make for a good addon, it certainly is a logical thing to control with trust.

**Comment by RoboMWM** [comment]
Hmm, some day when I have time I'll sift through BigScary's responses to these age-old questions and put 'em up in a FAQ.

**Comment by RoboMWM** [comment, marked as answer]
> I fail to see how not protecting pressure plates and especially buttons consistent, it always seemed like a "thats weird, oh well" kind of thing. Everything should be protected with a switch in the config file to turn off if you wish

Answered in the FAQ (which is gonna be mostly a compilation of Big_Scary's responses, who made most of the design decisions as the creator of the plugin).

Unsure if there's something for redstone, but this is a pretty niche case where someone's intentionally putting devices on the border of their claim... And would make it complicated to determine if a signal should or should not pass over a claim border (i.e. case where two claims owned by the same person are adjacent to each other.)
